### PR TITLE
Add test for header middleware

### PR DESCRIPTION
**Pull request by Ashwithabg**
#45 

Co-authored-by: Ravi <ravikumarkarn080@gmail.com>

### CODE REVIEW

File: handle/middlewares/headers_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const CookieHeader = "Set-Cookie"
+
+func TestAPIHeader(t *testing.T) {
+	now = func() time.Time {
+		return time.Date(2020, time.October, 9, 1, 2, 3, 4, time.UTC)
+	}
+
+	url := "http://localhost:3000/test"
+	expectedCookie := "gocity_api=http://localhost:3000/test; Path=/; Expires=Sat, 10 Oct 2020 01:02:03 GMT; Max-Age=90000"
```
Comment (rodrigo-brito) [inline handle/middlewares/headers_test.go:19]: A better way to check cookies from recorder:

```
  request := &http.Request{Header: http.Header{"Cookie": recorder.HeaderMap["Set-Cookie"]}}
  cookie, err := request.Cookie(cookieKey)
  require.NoError(t, err)

  require.Equal(t, cookie.Value, "<URL HERE>")
```

File: handle/middlewares/headers.go [hunk 2]
```diff
@@ -8,9 +8,10 @@ import (
 const cookieKey = "gocity_api"
 
 type Middleware func(http.Handler) http.Handler
+var now = time.Now
```
[the remark this answers is not in the document]
Comment (rodrigo-brito) [reply handle/middlewares/headers.go:11]: In this case, you can get the expiration from the cookie and check if the value is greater than a specific value. Or use a threshold. Avoid using strings to check cookies. A better assertion is described below.

File: handle/middlewares/headers_test.go (hunk 1 above)
Comment (Ashwithabg) [reply handle/middlewares/headers_test.go]: Have done the changes.
Could you please review

File: handle/middlewares/headers_test.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+func TestAPIHeader(t *testing.T) {
+	url := "http://localhost:3000/test"
+	middleware := APIHeader(url)
+	recorder := httptest.NewRecorder()
+
+	router := middleware(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
+	router.ServeHTTP(recorder, nil)
+
+	request := &http.Request{Header: http.Header{"Cookie": recorder.HeaderMap["Set-Cookie"]}}
```
Comment (rodrigo-brito) [inline handle/middlewares/headers_test.go:19]: The linter identifies `HeaderMap` as deprecated. Maybe `recorder.Header()` is a good alternative?

**Comment by rodrigo-brito** [comment]
It looks great! Thank you for your contribution!

**Comment by Ashwithabg** [comment]
Thanks @rodrigo-brito 
It was a good learning for me.

**Pull request merged**